samples/net/sockets/coap_service/src/test.c

```diff
@@ -0,0 +1,262 @@
+
+	if (type == COAP_TYPE_CON) {
+		type = COAP_TYPE_ACK;
+	} else {
+		type = COAP_TYPE_NON_CON;
+	}
+
+	r = coap_packet_init(&response, data, sizeof(data),
+			     COAP_VERSION_1, type, tkl, token,
+			     COAP_RESPONSE_CODE_DELETED, id);
+	if (r < 0) {
+		return r;
+	}
+
+	r = coap_resource_send(resource, &response, addr, addr_len);
+
+	return r;
+}
+
+static int test_put(struct coap_resource *resource,
+		    struct coap_packet *request,
+		    struct sockaddr *addr, socklen_t addr_len)
+{
+	uint8_t data[CONFIG_COAP_SERVER_MESSAGE_SIZE];
+	struct coap_packet response;
+	uint8_t token[COAP_TOKEN_MAX_LEN];
+	const uint8_t *payload;
+	uint16_t payload_len;
+	uint8_t code;
+	uint8_t type;
+	uint8_t tkl;
+	uint16_t id;
+	int r;
+
+	code = coap_header_get_code(request);
+	type = coap_header_get_type(request);
+	id = coap_header_get_id(request);
+	tkl = coap_header_get_token(request, token);
+
+	LOG_INF("*******");
+	LOG_INF("type: %u code %u id %u", type, code, id);
+	LOG_INF("*******");
+
+	payload = coap_packet_get_payload(request, &payload_len);
+	if (payload) {
+		net_hexdump("PUT Payload", payload, payload_len);
+	}
+
+	if (type == COAP_TYPE_CON) {
+		type = COAP_TYPE_ACK;
+	} else {
+		type = COAP_TYPE_NON_CON;
+	}
+
+	r = coap_packet_init(&response, data, sizeof(data),
+			     COAP_VERSION_1, type, tkl, token,
+			     COAP_RESPONSE_CODE_CHANGED, id);
+	if (r < 0) {
+		return r;
+	}
+
+	r = coap_resource_send(resource, &response, addr, addr_len);
+
+	return r;
+}
+
+static int test_post(struct coap_resource *resource,
+		     struct coap_packet *request,
+		     struct sockaddr *addr, socklen_t addr_len)
+{
+	static const char * const location_path[] = { "location1",
+						      "location2",
+						      "location3",
+						      NULL };
+	uint8_t data[CONFIG_COAP_SERVER_MESSAGE_SIZE];
+	const char * const *p;
+	struct coap_packet response;
+	uint8_t token[COAP_TOKEN_MAX_LEN];
+	const uint8_t *payload;
+	uint16_t payload_len;
+	uint8_t code;
+	uint8_t type;
+	uint8_t tkl;
+	uint16_t id;
+	int r;
+
+	code = coap_header_get_code(request);
+	type = coap_header_get_type(request);
+	id = coap_header_get_id(request);
+	tkl = coap_header_get_token(request, token);
+
+	LOG_INF("*******");
+	LOG_INF("type: %u code %u id %u", type, code, id);
+	LOG_INF("*******");
+
+	payload = coap_packet_get_payload(request, &payload_len);
+	if (payload) {
+		net_hexdump("POST Payload", payload, payload_len);
+	}
+
+	if (type == COAP_TYPE_CON) {
+		type = COAP_TYPE_ACK;
+	} else {
+		type = COAP_TYPE_NON_CON;
+	}
+
+	r = coap_packet_init(&response, data, sizeof(data),
+			     COAP_VERSION_1, type, tkl, token,
+			     COAP_RESPONSE_CODE_CREATED, id);
+	if (r < 0) {
+		return r;
+	}
+
+	for (p = location_path; *p; p++) {
+		r = coap_packet_append_option(&response,
+					      COAP_OPTION_LOCATION_PATH,
+					      *p, strlen(*p));
+		if (r < 0) {
+			return r;
+		}
+	}
+
+	r = coap_resource_send(resource, &response, addr, addr_len);
+
+	return r;
```

Comment:
Just pondering here, the functions look pretty similar to me. Would it be possible to refactor them to share the common code?

Reply:
This code was copied from the current `coap_server` sample with minimal changes, to prove the backwards compatibility.